### DISCUSSION TITLE
Is there any directory of themes?

### DISCUSSION THREAD

**Discussion by AllanXu49** [statiqdev/Discussions#148] [Q&A]
Greetings,

So far I found ( https://github.com/statiqdev/CleanBlog ) theme for Statiq. Is there any directory of available themes for Statiq?

Thank you,

**Comment by StevenTCramer** [comment]
To get ideas I used the following search to find other open source projects.

https://sourcegraph.com/search?q=context:global+file:csproj+Statiq.&patternType=literal

You can look at https://github.com/TheFreezeTeam/TheFreezeTeamBlog to see how we are using it for our blog.

**Comment by daveaglick** [comment]
Unfortunately not, and there aren't many generally available at the moment. I've had it on my long to-do list to port some of the Wyam themes over, and there’s a bunch from [HTML5 Up](https://html5up.net/) that would be good to get ported too, but just not enough hours available to do the work yet. The good news is that converting plain HTML themes for use with Statiq is pretty straightforward, so when I do get some time to do it I expect we'll get a flood of several new ones all at once. A bunch of folks are doing good work with bespoke themes that might be good examples though. I love @StevenTCramer search for finding ones on GitHub. This is also a good post that describes Phil's theme and how he built it: https://thirty25.com/posts/2022/05/tour-of-the-thirty25.com-repository

Longer-term I've also got plans for a "gallery" page on the Statiq website that showcases themes and other extensions, both official and community developed.

**Reply by AlexHedley** [reply]
Is there a list of Wyam themes anywhere?

I don't see any at https://github.com/Wyamio

**Reply by daveaglick** [reply]
There's a list of old Wyam themes here: https://wyam.io/recipes/blog/themes/. Because Statiq is an evolution of Wyam (and not just a new version) these most certainly won't work out of the box, though they might give a good starting place. If you're just looking for general HTML themes that can be adapted, I've had really good success with the ones at https://html5up.net/ and plan to port as many as I can to Statiq eventually.

**Reply by AlexHedley** [reply]
Might have a look at porting some myself.

Presume I can just use [CleanBlog](https://github.com/statiqdev/CleanBlog) as an example of setup etc and use most of the content from the [Blog](https://github.com/Wyamio/Wyam/tree/develop/themes/Blog) folder.

**Reply by daveaglick** [reply]
Yeah, CleanBlog is absolutely the place to start. I wrote it pretty early on mainly so that there would be an example of how to port other themes, even if just for myself. Never intended for it to be the only official one for so long, but you know how that goes.

**Reply by AlexHedley** [reply]
I've created blank repos as a start for each of the existing Themes, I'll work on these over time and update here when they're in a useable state, happy to give collaborator Writes or move to the StatiqDev org if that's easier to manage.

- https://github.com/AlexHedley/Phantom
- https://github.com/AlexHedley/SolidState
- https://github.com/AlexHedley/Stellar
- https://github.com/AlexHedley/Trophy
- https://github.com/AlexHedley/Samson (Docs)

**Reply by daveaglick** [reply]
That's amazing! No concerns here about having them under your own account - I'd love to get some more community-driven stuff out there. And I'm happy to go through a normal PR process :)

**Reply by suchja** [reply]
I've just started porting the Dopetrope theme from HTML5 UP to Statiq. Although there is a ton more to do, an initial version can be tested at [suchja/statiq-dopetrope-theme](https://github.com/suchja/statiq-dopetrope-theme). I'm trying to provide a [demo](https://github.com/suchja/statiq-dopetrope-demo) as well, but that has still some problems when trying to deploy on GitHub Pages. Locally it works like a charm and should give a good first impression of the theme.

**Reply by AlexHedley** [reply]
@suchja heres an example of my deployment wf [main.yml](https://github.com/alex-hedley/statiqweb-example/blob/main/.github/workflows/main.yml) and my [appsettings.json](https://github.com/alex-hedley/statiqweb-example/blob/main/src/appsettings.json)

**Reply by AlexHedley** [reply]
Looks like you have the source in your `gh-pages` branch.
Just go to Settings | Pages
Pick that branch
![](https://docs.github.com/assets/cb-24510/mw-1440/images/help/pages/publishing-source-folder-drop-down.webp)
Save
A new pages deployment action should show.

- https://suchja.github.io/statiq-dopetrope-demo

https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site

**Reply by suchja** [reply]
@AlexHedley thanks for the comprehensive feedback. Generally it works to choose the branch. However, GitHub states that in the few months all GitHub Pages will be deployed from GitHub Actions. So I was trying to use the GitHub Actions. I guess there is something missing in my GitHub Action. I see in your `main.yml` that you are using a specific deploy action for GitHub Pages. Maybe this is the missing part for me?

PS: In case you have even more time to spare 😇, maybe you have a quick answer to my [other discussion question](https://github.com/orgs/statiqdev/discussions/221)?

**Reply by AlexHedley** [reply]
I think I've used [configure-pages](https://github.com/actions/configure-pages) in another repo, I'll have a look.

I jump between
- https://github.com/JamesIves/github-pages-deploy-action
- https://github.com/peaceiris/actions-gh-pages

**Reply by daveaglick** [reply]
Go team! Love that you all are helping each other out since I obviously seem to be having a tough time of that lately :)

**Reply by AlexHedley** [reply]
Starter Template: [static.yml](https://github.com/actions/starter-workflows/blob/main/pages/static.yml)

**Reply by AlexHedley** [reply]
- https://github.com/ociaw/Verdoct

**Reply by AlexHedley** [reply]
_Legacy_

Wyam - **PlainBlog**

- https://github.com/ociaw/PlainBlog
